tests/templates/test_subroutines.py

```diff
@@ -926,5 +912,7 @@ def test_evolution_output(self, time, hamiltonian, steps, expectation):
         def circuit():
             ApproxTimeEvolution(hamiltonian, time, steps)
             return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_wires)]
-
+        print(circuit())
+        print(circuit.draw())
+        print(expectation)
```

Comment:
Are you using the code itself to design the test cases?

Reply:
Oops, I thought I deleted that

Reply:
Question still holds. How did you make the new test cases? Hopefully worked out by hand?

It's really not good practice to use the code to generate test cases, since it by definition won't find :beetle: :bug:

Reply:
I didn't work it out by hand: I manually made a PL circuit where I applied every gate one would expect would be applied by ApproxTimeEvolution, and got the results.

Reply:
Is that Ok?

Reply:
:sweat_smile: Got it! 

If that's the case, I'd recommend a "comparison" test that does exactly what you did, but explicitly (easier for future developers to see where the expected answers in a test come from)